advisor/src/main/kotlin/advisors/NexusIq.kt

```diff
@@ -154,9 +153,9 @@ class NexusIq(
 
     private fun NexusIqService.SecurityIssue.toVulnerability(): Vulnerability {
         val browseUrl = if (url == null && reference.startsWith("sonatype-")) {
-            URL("${nexusIqConfig.browseUrl}/assets/index.html#/vulnerabilities/$reference")
+            "${nexusIqConfig.browseUrl}/assets/index.html#/vulnerabilities/$reference"
         } else {
-            url
+            url.toString()
```

Comment:
*If* we make `url` a String in `Vulnerability`, shouldn't it then also become a String in `SecurityIssue`? Or would your argument be that `SecurityIssue` is Nexus-IQ-specific where we really know it to be a URL, and here we just transform the URL to a more generic representation in the class that is common across all providers?

Reply:
I would tend to change the type in `SecurityIssue` as well. @MarcelBochtler, what do you think?

Reply:
As your arguments also apply for the `url` in `SecurityIssue`, I think changing it there as well would be a good idea.